### PR TITLE
CMakeLists.txt: Add the option OSX_FRAMEWORK to control OSX builds.

### PR DESCRIPTION
**Pull request by AlexandraTrifan**
On MacOS we should be able to build both frameworks and dylibs.

Signed-off-by: Alexandra Trifan <Alexandra.Trifan@analog.com>

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -468,11 +469,16 @@ if (WIN32)
...
+if(NOT OSX_FRAMEWORK)
+	set(FRAMEWORK_ON False)
+endif()
+
 add_library(iio ${LIBIIO_CFILES} ${LIBIIO_HEADERS} ${LIBIIO_EXTRA_HEADERS} ${LIBIIO_RC})
 set_target_properties(iio PROPERTIES
 	VERSION ${VERSION}
 	SOVERSION ${LIBIIO_VERSION_MAJOR}
-	FRAMEWORK TRUE
+	FRAMEWORK ${FRAMEWORK_ON}
```
Comment (pcercuei) [inline CMakeLists.txt:481]: `FRAMEWORK ${OSX_FRAMEWORK}` does not work?

File: CMakeLists.txt [hunk 2]
```diff
@@ -2,6 +2,7 @@ message(STATUS "cmake version: ${CMAKE_VERSION}")
 cmake_minimum_required(VERSION 2.8.7)
 project(libiio C)
 
+option(OSX_FRAMEWORK "Create a OSX_FRAMEWORK" ON)
```
Comment (pcercuei) [inline CMakeLists.txt:5]: I'd prefer the option to be move just above where it's used for the first time.

File: CMakeLists.txt [hunk 3]
```diff
@@ -14,7 +15,7 @@ if (WIN32)
 endif()
 
 # Set the default install path to /usr
-if (NOT WIN32 AND CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+if (NOT APPLE AND NOT WIN32 AND CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
```
Comment (pcercuei) [inline CMakeLists.txt:18]: Why the `NOT APPLE`? Are non-framework builds expected to install in /usr/local? There is no info about this on the commit's message.

Comment (AlexandraTrifan) [reply CMakeLists.txt:18]: Installing directly into /usr/lib or /usr/bin on MacOS raises an error. I think it's due to System Integrity Protection.

**Review by pcercuei — APPROVED** [review]

**Comment by pcercuei** [comment]
Merged, thanks!

**Pull request merged**